documentation/specs/NexusLayout.md

```diff
@@ -0,0 +1,465 @@
+# `NexusLayout` Component Specification
+
+## Overview
+
+`NexusLayout` defines the header, main content, and multi-page content areas of a Nexus product page.
+
+### Prior Art
+
+- [Primer `<PageLayout />`](https://primer.style/design/components/page-layout/react)
+- [Paste `<SidebarNavigation /`>](https://paste.twilio.design/components/sidebar-navigation)
+
+---
+
+## Design
+
+`NexusLayout` will be a compound component consisting of `NexusLayout`, `NexusLayout.Header`, `NexusLayout.Content`, and `NexusLayout.MultipageContainer`. `NexusLayout` also provides components for building navigation with `NexusLayout.Nav` and `NexusLayout.SidebarNav`.
+
+`NexusLayout` is highly composable. Subcomponents within a `NexusLayout` can be replaced as needed. Subcomponents are lightweight wrappers with built-in styles and constraints.
+
+`NexusLayout` is concerned only with presentational structure. It is meant to be wrapped by an app layout that may include app-specific business logic and configuration.
+
+`NexusLayout` can render an empty content container with `NexusLayout.Content`. It can render a multi-page navigational layout with `NexusLayout.MultipageContainer`.
+
+### API
+
+```ts
+type NexusLayoutProps = {
+  children: ReactNode;
+};
+
+type NexusLayoutHeaderProps = {
+  /** Optional logo to render if different than EasyPost */
+  logo?: ReactNode;
+  children: ReactNode;
+};
+
+type NexusLayoutContentProps = {
+  children: ReactNode;
+};
+
+// Header navigation components
+
+type NexusLayoutNavProps = {
+  /**
+   * Optional label for the navigation.
+   * @default Main
+   */
+  "aria-label"?: string;
+
+  /**
+   * The currently selected key in the navigation list.
+   */
+  selectedKey?: Key;
+
+  children: ReactNode;
+};
+
+type NexusLayoutNavLinkProps = {
+  /** Icon for nav link */
+  iconSymbol: IconSymbol;
+
+  /** Key for the link. Usually the href */
+  key: Key;
+} & ComponentProps<"a">;
+
+// Header action components (buttons on the right side of header)
+
+type NexusLayoutActionsProps = {
+  children: ReactNode;
+};
+
+type NexusLayoutMenuActionProps = {
+  /** Optional custom accessibility label describing the menu action. */
+  accessibilityLabel?: string;
+
+  /** Icon symbol for the action. */
+  iconSymbol: IconSymbol;
+
+  /** Slot for badge to be rendered. */
+  renderBadge?: () => ReactNode;
+
+  /** Slot for rendering the menu overlay. */
+  children: ReactNode;
+};
+
+// Multi-page components
+
+type NexusLayoutMultipageContainerProps = {
```

Comment:
not sure i love the name `Multipage` for the nested navigation pages but it what was called in Figma. open to suggestions.

Reply:
i cant think of a better name off the top of my head. i was thinking maybe something like `NexusLayoutSectionalContainer` but i think then we'd end up with stuff like `...SectionalSidebar` which sounds a bit odd lol. i think its fine as is